### PR TITLE
Add a lint for casts from char literals to u8

### LINKED ISSUE #563: Lint `<char literal> as u8`
[This](https://internals.rust-lang.org/t/to-reduce-the-number-of-true-casts/3076) argues that the use of `as` casts should be minimised for safety reasons. One type of cast that can easily be avoided but is quite common in rust is casting from a char literal to `u8`. If the char literal is ascii, you can avoid the cast by using a byte literal (e.g. `b'a'`) instead, and if the char literal is non-ascii, the code is probably not doing what the author intended anyway [citation needed]. Therefore I suggest we lint all casts from char literals to `u8`, and suggest using byte literals instead.

### PR DESCRIPTION
**Pull request by fhartwig**
Fixes #563

### CODE REVIEW

**Comment by llogiq** [comment]
Strange that README isn't updated.

File: src/types.rs [hunk 1]
```diff
@@ -517,3 +517,41 @@ impl<'v> Visitor<'v> for TypeComplexityVisitor {
...
+        use syntax::ast::{Lit_, UintTy};
+
+        if let ExprCast(ref e, _) = expr.node {
+            if let ExprLit(ref l) = e.node {
+                if let Lit_::LitChar(_) = l.node {
+                    if ty::TyUint(UintTy::TyU8) == cx.tcx.expr_ty(expr).sty && !in_macro(cx, expr.span) {
+                        let msg = "casting character literal to u8.";
+                        let help = format!("Consider using a byte literal \
+                                            instead:\nb{}",
+                                          snippet(cx, e.span, "'x'"));
```
Comment (Manishearth) [inline src/types.rs]: Add a note here mentioning that `char`s are 4 bytes wide in Rust.

**Comment by fhartwig** [comment]
@llogiq I did run the update script, but it doesn't seem to do anything. I'll look into it.

**Comment by fhartwig** [comment]
@llogiq Regex didn't like the digit in the lint name. I updated the update script.

File: src/types.rs (hunk 1 above)
Comment (fhartwig) [inline src/types.rs]: Done.

**Pull request merged**